compiler/rustc_monomorphize/src/collector.rs

```diff
@@ -923,6 +924,17 @@ fn visit_instance_use<'tcx>(
         return;
     }
 
+    if let ty::InstanceDef::Intrinsic(def_id) = instance.def {
+        let name = tcx.item_name(def_id);
+        if let Some(_requirement) = ValidityRequirement::from_intrinsic(name) {
+            let def_id = tcx.lang_items().get(LangItem::PanicNounwind).unwrap();
```

Comment:
A comment on this block, possibly mentioning `assert_inhabited`, `assert_zero_valid`, and `assert_mem_uniniitalized_valid`, would help a lot.